src/transformers/generation/candidate_generator.py

```diff
@@ -175,6 +182,7 @@ def get_candidates(self, input_ids: torch.LongTensor) -> Tuple[torch.LongTensor,
         # Don't generate more than `max_length - 1` candidates since the target model generates one extra token.
         new_cur_len = input_ids.shape[-1]
         max_new_tokens = min(int(self.num_assistant_tokens), self.generation_config.max_length - new_cur_len - 1)
+        min_new_tokens = min(max_new_tokens, self.min_length - new_cur_len)
```

Comment:
Why use a class attribute and not the generation config, as for `max_length`

Reply:
@zucchini-nlp it's the other way around, if anything we would want to deprecate the `min_tokens` argument/config option :) `max_new_tokens` and `min_new_tokens` are much more predictable from a user point of view, as the user doesn't need to be concerned with the input length. In the past, before `max_new_tokens` and `min_new_tokens` were introduced, we would often get issues from confused users. 

Inside generate, however, it is much easier to use the total length for control (no need to track the input length). We set `max_length` from `max_new_tokens` when needed ([here](https://github.com/huggingface/transformers/blob/d47966536cd5ac1ed7e140edac65f00f471f656f/src/transformers/generation/utils.py#L1419)), perhaps we should do the same with `min_length` to simplify the procedure in this PR :)

Reply:
I decided to make a separate method for all length related corrections. Added one more test for min length, same was we had for max length.